src/psyclone/tests/psyGen_test.py

```diff
@@ -2621,8 +2631,465 @@ def test_binaryoperation_can_be_printed():
+    assert "Unexpected kernel AST. Could not find " \
+           "subroutine: dummy_code" in str(error.value)
+
+    # Test corrupting ast by deleting specification part
+    del ast2.content[0].content[1]
+    with pytest.raises(InternalError) as error:
+        schedule = processor.generate_schedule("dummy_code", ast2)
+    assert "Unexpected kernel AST. Could not find specification part."
+
+
+def test_fparser2astprocessor_generate_schedule_two_modules():
+    ''' Tests the fparser2AST generate_schedule method with an simple
+    subroutine.
+    '''
+    ast1 = fpapi.parse(FAKE_KERNEL_METADATA*2, ignore_comments=True)
+    metadata = DynKernMetadata(ast1)
+    my_kern = DynKern()
+    my_kern.load_meta(metadata)
+    ast2 = my_kern.ast
+    processor = Fparser2ASTProcessor()
+
+    # Test properly formed kernel module
+    with pytest.raises(InternalError) as error:
+        schedule = processor.generate_schedule("dummy_code", ast2)
+    assert ("Unexpected kernel AST. Just one module definition per "
+            "file supported.") in str(error.value)
+
+
+def test_fparser2astprocessor_generate_schedule_dummy_subroutine():
+    ''' Tests the fparser2AST generate_schedule method with an simple
+    subroutine.
+    '''
+    DUMMY_KERNEL_METADATA = '''
+    module dummy_mod
+      type, extends(kernel_type) :: dummy_type
+         type(arg_type), meta_args(3) =                    &
+              (/ arg_type(gh_field, gh_write,     w3),     &
+                 arg_type(gh_field, gh_readwrite, wtheta), &
+                 arg_type(gh_field, gh_inc,       w1)      &
+               /)
+         integer :: iterates_over = cells
+       contains
+         procedure, nopass :: code => dummy_code
+      end type dummy_type
+    contains
+     subroutine dummy_code(f1, f2, f3)
+        real(wp), dimension(:,:), intent(in)  :: f1
+        real(wp), dimension(:,:), intent(out)  :: f2
+        real(wp), dimension(:,:) :: f3
+        f2 = f1 + 1
+      end subroutine dummy_code
+    end module dummy_mod
+    '''
+    ast1 = fpapi.parse(DUMMY_KERNEL_METADATA, ignore_comments=True)
+    metadata = DynKernMetadata(ast1)
+    my_kern = DynKern()
+    my_kern.load_meta(metadata)
+    ast2 = my_kern.ast
+    processor = Fparser2ASTProcessor()
+
+    # Test properly formed kernel module
+    schedule = processor.generate_schedule("dummy_code", ast2)
+    assert isinstance(schedule, KernelSchedule)
+
+    # Test argument intent is infered when not available in the declaration
+    assert schedule.symbol_table.lookup('f3').access == 'readwrite_arg'
+
+    # Test corrupting ast by deleting execution part
+    del ast2.content[0].content[2].content[1].content[2]
+    schedule = processor.generate_schedule("dummy_code", ast2)
+    assert isinstance(schedule, KernelSchedule)
+
+
+def test_fparser2astprocessor_generate_schedule_unmatching_arguments():
+    ''' Tests the fparser2AST generate_schedule method with an simple
```

Comment:
Please correct this docstring to describe the test.

Reply:
Done